app/harvesters/hathi_harvester.rb

```diff
@@ -0,0 +1,104 @@
+require 'net/http'
+require 'rubygems/package'
+require 'zlib'
+
+
+class HathiHarvester < Krikri::Harvesters::MarcXMLHarvester
+  BASE_URL = 'http://mirlyn-aleph.lib.umich.edu/mdp/dpla_metadata/'
+
+  def initialize(opts = {})
+    if !(opts[:url] || opts[:dump_file])
+      raise "You must provide either `:url` or `:dump_file`"
+    end
+
+    opts[:uri] = (opts[:url] || opts[:dump_file])
```

Comment:
> In tarball_source we'll need to make sure to call close! on the result of download_url if that branch gets hit (otherwise the temp file sticks around, and they're not small files ;) ).

:+1:. It might even be a good idea to just use `File.open(path, mode, &block)` for this. That provides a pretty common interface; but maybe complicates the URL tempfile logic a bit.

> Happy to have a go at making these changes.

:rocket: that would be awesome.